src/jit/emitxarch.cpp

```diff
@@ -13663,6 +13663,1128 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+            result.insLatency = PERFSCORE_LATENCY_ZERO;
+            switch (insFmt)
+            {
+                case IF_LABEL:
+                    result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+                    break;
+
+                case IF_METHOD:
+                    result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+                    break;
+
+                case IF_METHPTR:
+                    result.insThroughput = PERFSCORE_THROUGHPUT_3C;
+                    break;
+
+                case IF_SRD:
+                    result.insThroughput = PERFSCORE_THROUGHPUT_3C;
+                    break;
+
+                case IF_ARD:
+                    result.insThroughput = PERFSCORE_THROUGHPUT_3C;
+                    break;
+
+                default:
+                    assert(!"Unhandled insFmt for INS_call");
+                    result.insThroughput = PERFSCORE_THROUGHPUT_DEFAULT;
+                    break;
+            }
+            break;
+
+        case INS_ret:
+            if (insFmt == IF_CNS)
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_2C;
+            }
+            else
+            {
+                assert(insFmt == IF_NONE);
+                result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            }
+            break;
+
+        case INS_lock:
+            result.insThroughput = PERFSCORE_THROUGHPUT_13C;
+            break;
+
+        case INS_xadd:
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            break;
+
+        case INS_cmpxchg:
+            result.insThroughput = PERFSCORE_THROUGHPUT_5C;
+            break;
+
+        case INS_xchg:
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            if (memFmt == IF_NONE)
+            {
+                result.insLatency = PERFSCORE_LATENCY_1C;
+            }
+            else
+            {
+                result.insLatency = PERFSCORE_LATENCY_23C;
+            }
+            break;
+
+#ifdef _TARGET_X86_
+        case INS_fld:
+        case INS_fstp:
+            result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+            if (memAccessKind == PERFSCORE_MEMORY_NONE)
+            {
+                result.insLatency = PERFSCORE_LATENCY_1C;
+            }
+            break;
+#endif // _TARGET_X86
+
+#ifdef _TARGET_AMD64_
+        case INS_movsq:
+        case INS_stosq:
+#endif // _TARGET_AMD64_
+        case INS_movsd:
+        case INS_stosd:
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            break;
+
+#ifdef _TARGET_AMD64_
+        case INS_r_movsq:
+        case INS_r_stosq:
+#endif // _TARGET_AMD64_
+        case INS_r_movsd:
+        case INS_r_movsb:
+        case INS_r_stosd:
+        case INS_r_stosb:
+            // Actually variable sized: rep stosd, used to zero frame slots
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_25C;
```

Comment:
Why 25c?

Should `stosd`/`movsd` be modeled separately given they aren't applicable for the `Enhanced REP MOVSB` support?

Reply:
The actual cost is variable depending upon how many times the instruction repeats, (which is based on the value loaded into RCX)